pkg/hubble/metrics/api/context.go

```diff
@@ -281,10 +404,30 @@ func destinationIPContext(flow *pb.Flow) (context string) {
 	return
 }
 
+func getK8sAppFromLabels(labels []string) string {
+	ls := ciliumLabels.ParseLabelArrayFromArray(labels)
```

Comment:
This is unfortunately a bit of an expensive operation, and we've had to optimize similar code in the past.

Have you considered doing a `strings.HasPrefix` on `podAppLabels` and only parse the label if the prefix matches?

Reply:
I figured re-using the code we already had was best. Is there not any less-expensive option available in the labels package we can use?

Reply:
I don't think there is an existing function for this. Hubble's use of `[]string` predates the integration with the Cilium codebase, which typically uses `labels.LabelArray` as the in-memory type.